toolchain/check/impl_lookup.cpp

```diff
@@ -130,6 +131,38 @@ auto LookupImplWitness(Context& context, SemIR::LocId loc_id,
     }
   }
 
+  auto& stack = context.impl_lookup_stack();
```

Comment:
LookupImplWitness is getting long with these additions, had you considered moving the bottom for loop out to a helper function so that you wouldn't need a scope exit, or similar refactorings?